### ISSUE TITLE
selenium-server-standalone-3.0.1 can not find firefox binary configured in Json

### ISSUE THREAD

**Issue by tomMantsch** [SeleniumHQ/selenium#3123]
## Meta -
OS:  Windows 7
Selenium Version:  3.0.1
Browser:  Firefox
Browser Version:  49.0.2

## Expected Behavior - Before upgrading to selenium-server-standalone-3.0.1 I was running multiple Firefox ESR versions on my grid node with selenium-server-standalone-2.45.0. The configuration in JSON looks like this 

`{
          "browserName": "firefox",
          "version": "49.0.2",
          "platform": "WINDOWS",
		  "firefox_binary": "C:\Program Files (x86)\Mozilla Firefox\ESR\Firefox_49.0.2\firefox.exe",
          "maxInstances": 5,
		  "seleniumProtocol": "WebDriver"
        },`

Firefox 49.0.2 is starting and I can use multiple firefox version on my grid node.

## Actual Behavior -

Exception: Cannot find firefox binary in PATH. Make sure firefox is installed.

When I install Firefox in C:\Program Files (x86)\Mozilla Firefox which is the default location it works fine and firefox is starting.

## Steps to reproduce -
`{
          "browserName": "firefox",
          "version": "49.0.2",
          "platform": "WINDOWS",
		  "firefox_binary": "C:\Program Files (x86)\Mozilla Firefox\ESR\Firefox_49.0.2\firefox.exe",
          "maxInstances": 5,
		  "seleniumProtocol": "WebDriver"
        },`

**Comment by tomMantsch** [comment]
I just found this issue https://github.com/SeleniumHQ/selenium/issues/2948

What is the correct supposed way to support multiple firefox versions on a selenium grid node?

**Comment by hheath** [comment]
Ok, the workaround is to specify all versions of firefox on the startup command line, but that's an extremely messy way of doing things, surely?  Isn't that why the config .json file is there for, so that you can simplify the startup line and put all variables that need passing into the Grid node startup into the config file?

**Comment by mach6** [comment]
Is this issue still present for you on a newer release? 

I can't seem to reproduce it using 3.5.3 -- with this example `nodeConfig.json`

```json
{
  "capabilities": [{
    "browserName": "firefox",
    "acceptSslCerts": true,
    "javascriptEnabled": true,
    "takesScreenshot": true,
    "firefox_binary": "/foo/bin/firefox51",
    "maxInstances": 15,
    "version": 50
  }, {
    "browserName": "firefox",
    "acceptSslCerts": true,
    "javascriptEnabled": true,
    "takesScreenshot": true,
    "firefox_binary": "/bar/bin/firefox52",
    "maxInstances": 15,
    "version": 52
  }],
  "register": true,
  "hub": "http://localhost:4444/grid/register"
}
```

The node registers with the hub and has the following capabilities displayed in the web console;

```
capabilities: Capabilities [{firefox_binary=/foo/bin/firefox51, seleniumProtocol=WebDriver, acceptSslCerts=true, browserName=firefox, takesScreenshot=true, javascriptEnabled=true, maxInstances=15, version=50, platform=MAC}]
capabilities: Capabilities [{firefox_binary=/bar/bin/firefox52, seleniumProtocol=WebDriver, acceptSslCerts=true, browserName=firefox, takesScreenshot=true, javascriptEnabled=true, maxInstances=15, version=52, platform=MAC}]
```

@barancev @shs96c fyi

**Comment by shs96c** [comment]
@mach6 the node is registered correctly, but the capabilities aren't passed through when we start the session, so the actual firefoxdriver instance has the default binary path.

**Comment by ksiv** [comment]
Should I use "firefox_binary" in nodeconfig.json ? But a "binary" in java bindings? and "firefox.bin" in command line? Can anyone point me to a documentation where all the magic explained?
btw. Same issue with FF and I m solving this by passing capabilities from client.
`capabilities.setCapability(FirefoxDriver.BINARY,"C:\\ff46\\firefox.exe"); `

**Comment by shs96c** [comment]
This bug has been fixed in ae02d81442df2949e8d4de4f40b51b6181d301d8 and is part of 3.7
